crates/sage/src/scoring.rs

```diff
@@ -53,6 +54,7 @@ impl AddAssign<InitialHits> for InitialHits {
 /// Features of a candidate peptide spectrum match
 pub struct Feature {
     #[serde(skip_serializing)]
+    pub psm_id: String,
```

Comment:
Do you have any benchmarking data on the memory and speed impact of generating UUID v4's in the middle of scoring?

TBH I don't like the idea and it seems terribly wasteful, considering a monotonic counter will accomplish the same task, and could also just be done when writing results in an allocation-free manner

Reply:
Thank you for providing the suggestion. I have implemented a self-incrementing counter and removed the use of UUID v4.